homeassistant/components/jewish_calendar/service.py

```diff
@@ -17,29 +18,56 @@
 )
 from homeassistant.helpers import config_validation as cv
 from homeassistant.helpers.selector import LanguageSelector, LanguageSelectorConfig
+from homeassistant.helpers.sun import get_astral_event_date
+from homeassistant.util import dt as dt_util
 
-from .const import ATTR_DATE, ATTR_NUSACH, DOMAIN, SERVICE_COUNT_OMER
+from .const import AFTER_SUNSET, ATTR_DATE, ATTR_NUSACH, DOMAIN, SERVICE_COUNT_OMER
 
+_LOGGER = logging.getLogger(__name__)
 SUPPORTED_LANGUAGES = {"en": "english", "fr": "french", "he": "hebrew"}
 OMER_SCHEMA = vol.Schema(
     {
-        vol.Required(ATTR_DATE, default=datetime.date.today): cv.date,
+        vol.Optional(ATTR_DATE): cv.date,
+        vol.Optional(AFTER_SUNSET, default=False): cv.boolean,
         vol.Required(ATTR_NUSACH, default="sfarad"): vol.In(
             [nusach.name.lower() for nusach in Nusach]
         ),
-        vol.Required(CONF_LANGUAGE, default="he"): LanguageSelector(
+        vol.Optional(CONF_LANGUAGE, default="he"): LanguageSelector(
             LanguageSelectorConfig(languages=list(SUPPORTED_LANGUAGES.keys()))
         ),
     }
 )
 
 
+# top level for mocking tests
+def get_hebrew_date(hass: HomeAssistant) -> HebrewDate:
```

Comment:
Maybe a cleaner way is to return today's date.
That way, the below logic becomes: 

```python
date = call.data.get("date", get_date())
after_sunset = call.data.get(AFTER_SUNSET, is_after_sunset())
hebrew_date = HebrewDate.from_gdate(date + timedelta(days=int(after_sunset))
```

This has an additional advantage to allow asking for after_sunset of today.
So we can have:
- a specific date 
- a specific date after sunset
- today (nothing specified)
- today after sunset (only after sunset specified)

The only "issue" is if someone asks after_sunset and it's after midnight where they probably meant simply today.